unittests/wasm_tests.cpp

```diff
@@ -686,6 +686,44 @@ BOOST_FIXTURE_TEST_CASE( table_init_tests, TESTER ) try {
 
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE( table_init_oob, TESTER ) try {
+   create_accounts( {N(tableinitoob)} );
+   produce_block();
+
+   signed_transaction trx;
+   trx.actions.emplace_back(vector<permission_level>{{N(tableinitoob),config::active_name}}, N(tableinitoob), N(), bytes{});
+   trx.actions[0].authorization = vector<permission_level>{{N(tableinitoob),config::active_name}};
+
+    auto pushit_and_expect_fail = [&]() {
+      produce_block();
+      trx.signatures.clear();
+      set_transaction_headers(trx);
+      trx.sign(get_private_key(N(tableinitoob), "active"), control->get_chain_id());
+      
+      //the unspecified_exception_code comes from WAVM, which manages to throw a WAVM specific exception
+      // up to where exec_one captures it and doesn't understand it
```

Comment:
can we catch it somewhere in the wavm plumbing and re-code it to a common `wasm_execution_error` when we still have a chance at understanding it? 

Allowing anything that `FC_RETHROW_EXCEPTIONS` doesn't understand seems very broad and I'd expect us to want to know if this moves on us.